project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionLocalTests.java

```diff
@@ -0,0 +1,431 @@
+
+        // Push package_c:0.0.1 to central
+        cacheDependencyToCentralRepository(PACKAGES_DIRECTORY.resolve("package_c_0_0_1"), projectEnvironmentBuilder);
+
+        // Change package_c with same version and push to local
+        cacheDependencyToLocalRepository(PACKAGES_DIRECTORY.resolve("package_c_0_0_1_v2"), projectEnvironmentBuilder);
+
+        // Push package_d:0.0.1 to local
+        // Note we are changing the package here as well to support the api change in package_c
+        cacheDependencyToLocalRepository(PACKAGES_DIRECTORY.resolve("package_d_0_0_1"), projectEnvironmentBuilder);
+
+        // Build project and assert deps toml
+        BuildProject buildProject = BuildProject.load(projectEnvironmentBuilder, projectDirPath);
+        buildProject.save();
+        failIfDiagnosticsExists(buildProject);
+
+        Assert.assertEquals(readFileAsString(projectDirPath.resolve(DEPENDENCIES_TOML)),
+                readFileAsString(projectDirPath.resolve(RESOURCE_DIR_NAME).resolve("Dependencies.toml")));
+    }
+
+    @Test(description = "Tests Local repository packages (published, but patched as a newer unpublished version) " +
+            "as indirect dependencies", dependsOnMethods = "testCase0004")
+    public void testCase0006(ITestContext ctx) throws IOException {
+        /*
+            Make changes to the api of package_c and pushed to local repository as version 0.0.3
+            Version 0.0.1 is in central
+            Build project_0006:
+              - Add package_c new version 0.0.3 to Ballerina.toml of project_0006. package_b and package_c both
+              resolved from local repo
+              - Remove package_c from Dependencies.toml of project_0006, Then trans dep package_c resolved from
+              central, even though it was mentioned as local dependency in package_b Ballerina.toml
+         */
+
+        Path projectDirPath = PROJECTS_DIRECTORY.resolve("project_0006");
+        ctx.getCurrentXmlTest().addParameter("packagePath", String.valueOf(projectDirPath));
+
+
+        // project_0006 ---> package_b:0.0.1 ---> package_c:0.0.3
+
+        // package_b:0_0_1 and package_c:0_0_1 pushed to local from previous test
+        // package_c:0_0_1 pushed to central from previous test
+
+        // Push package_c:0_0_3 to local
+        cacheDependencyToLocalRepository(PACKAGES_DIRECTORY.resolve("package_c_0_0_3"), projectEnvironmentBuilder);
+
+        // Build project and assert deps toml
+        BuildProject buildProject = BuildProject.load(projectEnvironmentBuilder, projectDirPath);
+        buildProject.save();
+        failIfDiagnosticsExists(buildProject);
+
+        Assert.assertEquals(readFileAsString(projectDirPath.resolve(DEPENDENCIES_TOML)),
+                readFileAsString(projectDirPath.resolve(RESOURCE_DIR_NAME).resolve("Dependencies.toml")));
+
+
+        // Change the Dependencies.toml
+        Files.deleteIfExists(projectDirPath.resolve(DEPENDENCIES_TOML));
```

Comment:
Why are we changing the Dependencies.toml? The edit operations should only be done to Ballerina.toml

Reply:
I thought that was part of the testing scenario since we have mentioned in the doc as such : 

```
Removed package_i from Dependencies.toml of package_a, Then trans dep package_i resolved from central, even though it was mentioned as local dependency in package_h Ballerina.toml
```

So i figured we had to remove it directly from the deps toml. 

Im assuming in this case, it should be from the `Ballerina.toml` instead of the `Dependencies.toml` right. 
Ill make the changes to the test case.